### ISSUE TITLE
Question: Transforming channels with RACSignals operators yields crash?

### ISSUE THREAD

**Issue by hokster7** [ReactiveCocoa/ReactiveCocoa#1177]
I ran into a pretty bizarre error whilst trying to use `RACChannel`s to express a kind of bi-directional multiplexer/demultiplexer, with no avail.
### What I want to achieve

I wanted a couple of sliders to be related to each other but not a direct 1-to-1 mapping.

The first, `numberSlider`, ranges from `0.f` to `1.0f`.
The second, `halfSlider` ranges from `0.f` to `0.5f`.

I want the values of these two sliders to be _mutually linked_ with the invariant that `numberSlider.value == halfSlider.value * 2.f` at all times. What it says on the tin, really.

This relationship should be upheld when I fiddle with either slider. This must not cause an infinite update loop.

i.e. if I slide `numberSlider` to `0.8`, then `halfSlider` is updated to `0.4`. And if I slide `halfSlider` now to `0.2`, then `numberSlider` is updated to `0.4`.

The easiest way to model this is with `RACChannel`. They're just `RACSignal`s, right? So clearly I should just `map:` the values with `/2` and `*2` (inverse operations of each other, forming an isomorphism for this channel) and be done with it.

But I get an error...
### Reproduction

The following code type-checks:

``` objc
    RACChannelTerminal *numberTerminal = [self.numberSlider rac_newValueChannelWithNilValue:[NSNumber numberWithFloat:0.f]];
    RACChannelTerminal *halfTerminal = [self.halfSlider rac_newValueChannelWithNilValue:[NSNumber numberWithInt:0]];

    [numberTerminal subscribe:[numberTerminal map:^(NSNumber *n){ // Can't subscribe...
        return [NSNumber numberWithFloat:[n floatValue] / 2];
    }]];
```

Stacktrace:

```
2014-03-17 01:14:46.286 BDDReactiveCocoa[40869:a0b] -[RACDynamicSignal didSubscribeWithDisposable:]: unrecognized selector sent to instance 0x9e97720
2014-03-17 01:14:46.288 BDDReactiveCocoa[40869:a0b] *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[RACDynamicSignal didSubscribeWithDisposable:]: unrecognized selector sent to instance 0x9e97720'
*** First throw call stack:
(
    0   CoreFoundation                      0x0183f5e4 __exceptionPreprocess + 180
    1   libobjc.A.dylib                     0x015c28b6 objc_exception_throw + 44
    2   CoreFoundation                      0x018dc903 -[NSObject(NSObject) doesNotRecognizeSelector:] + 275
    3   CoreFoundation                      0x0182f90b ___forwarding___ + 1019
    4   CoreFoundation                      0x0182f4ee _CF_forwarding_prep_0 + 14
    5   BDDReactiveCocoa                    0x000369cf -[RACPassthroughSubscriber initWithSubscriber:signal:disposable:] + 655
    6   BDDReactiveCocoa                    0x00077cf8 -[RACSubject subscribe:] + 424
    7   BDDReactiveCocoa                    0x00039984 __30-[RACReplaySubject subscribe:]_block_invoke + 1332
    8   BDDReactiveCocoa                    0x0007a338 -[RACSubscriptionScheduler schedule:] + 488
    9   BDDReactiveCocoa                    0x00039349 -[RACReplaySubject subscribe:] + 281
    10  BDDReactiveCocoa                    0x000252a7 -[RACChannelTerminal subscribe:] + 119
    11  BDDReactiveCocoa                    0x00003095 -[LOLMainViewController viewDidLoad] + 1077
    12  UIKit                               0x004489a8 -[UIViewController loadViewIfRequired] + 696
    13  UIKit                               0x00448c44 -[UIViewController view] + 35
    14  UIKit                               0x003715ad -[UIWindow addRootViewControllerViewIfPossible] + 66
    15  UIKit                               0x00371947 -[UIWindow _setHidden:forced:] + 312
    16  UIKit                               0x00371bdd -[UIWindow _orderFrontWithoutMakingKey] + 49
    17  UIKit                               0x0cf04c66 -[UIWindowAccessibility(SafeCategory) _orderFrontWithoutMakingKey] + 77
    18  UIKit                               0x0037c44a -[UIWindow makeKeyAndVisible] + 65
    19  UIKit                               0x0032f8e0 -[UIApplication _callInitializationDelegatesForURL:payload:suspended:] + 1851
    20  UIKit                               0x00333fb8 -[UIApplication _runWithURL:payload:launchOrientation:statusBarStyle:statusBarHidden:] + 824
    21  UIKit                               0x0034842c -[UIApplication handleEvent:withNewEvent:] + 3447
    22  UIKit                               0x00348999 -[UIApplication sendEvent:] + 85
    23  UIKit                               0x00335c35 _UIApplicationHandleEvent + 736
    24  GraphicsServices                    0x037922eb _PurpleEventCallback + 776
    25  GraphicsServices                    0x03791df6 PurpleEventCallback + 46
    26  CoreFoundation                      0x017badd5 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE1_PERFORM_FUNCTION__ + 53
    27  CoreFoundation                      0x017bab0b __CFRunLoopDoSource1 + 523
    28  CoreFoundation                      0x017e57ec __CFRunLoopRun + 2156
    29  CoreFoundation                      0x017e4b33 CFRunLoopRunSpecific + 467
    30  CoreFoundation                      0x017e494b CFRunLoopRunInMode + 123
    31  UIKit                               0x003336ed -[UIApplication _run] + 840
    32  UIKit                               0x0033594b UIApplicationMain + 1225
    33  BDDReactiveCocoa                    0x0000388d main + 141
    34  libdyld.dylib                       0x01e7b725 start + 0
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

Well. That shouldn't happen. Everything type-checked. The type-hierarchy says it's legal for me to `map:` and to then `subscribe:` a channel onto a mapped thing.

Suppose I implement the missing selector: what would happen...

```
// Add this code above the previous sample
@implementation RACDynamicSignal (HackAroundTheError)
- (void)didSubscribeWithDisposable:(RACDisposable *)disposable{
    // Do nothing!
}
@end
```

Then, the error becomes a bit more helpful:

```
2014-03-17 01:20:07.410 BDDReactiveCocoa[41063:a0b] -[RACDynamicSignal sendNext:]: unrecognized selector sent to instance 0x998b5f0
```

It looks like my program thinks `RACDynamicSignal` is a subscriber, but it isn't.

Basically, I'm just VERY confused about how `RACChannel`s were supposed to work in the first place. It may be brilliant for ensuring two ends of the channel hold the same value.
But can I use it to express slightly more useful bi-directional flows, where I transform the value in two (possibly different) ways, like dividing and multiplying by 2 in my case?

Can I have multiple channels sharing one central `RACChannelTerminal` in a star topology? How do I manipulate the signals between those? :/
